### ISSUE TITLE
Effects Icons do not display correct with Linux AppImage

### ISSUE THREAD

**Issue by MelissaFrog** [xLightsSequencer/xLights#4314]
Hello All,

First, I hope I am submitting this correctly.

The issue I am having if from both the 2023.01and the 2024.02 Linux AppImage releases.  On the sequence tab where the effects icons are displayed, the icons are not visible in the default (icon size) view and are distorted in other views.

Version 2023.23 worked fine in this respect.

I am running Ubuntu 20.04.6 LTS

![xLights_1](https://github.com/xLightsSequencer/xLights/assets/108493995/a62c9daf-8c04-4f49-9050-277bed1160a0)
![xLights_2](https://github.com/xLightsSequencer/xLights/assets/108493995/859809ea-0c71-4223-9f29-ec0d59a991ac)
![xLights_3](https://github.com/xLightsSequencer/xLights/assets/108493995/130e6f85-4e1a-4402-9167-2a4bbdf4c5de)

Regards,

Joe Hildreth

**Comment by derwin12** [comment]
At some point late last year they were linked to the resolution setting you had in windows. Perhaps that caused this issue?

**Comment by Gilimi-the-Grim** [comment]
Just as a note, my machine is setup with three screens each at 1920x1080.  I don't know if this matters.  I should also note I am new to xLights, but everything else that I have used in the software appears to be working.  It is just the effects Icons that ave gone awry. 

Regards,
Joe

**Comment by derwin12** [comment]
Just to clarify .. you said " from both the 2023.01and the 2024.02 " did you mean 2024.01 and .02?
So something happened between 2023.23 and 2024.01 ? That might help us narrow this down.

**Comment by AverageWaffleHouseEnjoyer** [comment]
> Just to clarify .. you said " from both the 2023.01and the 2024.02 " did you mean 2024.01 and .02? So something happened between 2023.23 and 2024.01 ? That might help us narrow this down.

I can reproduce this issue with a single screen setup at 1920x1080. It starts on 2024.01 and is not present in 2023.23 or earlier. Running from terminal produces many Gtk and GLib-GObject warnings / critical errors. Not sure if the errors are related to the effect icons or the blank windows issue from #4208 which seems to only happen on rolling release distros.

**Comment by Gilimi-the-Grim** [comment]
> Just to clarify .. you said " from both the 2023.01and the 2024.02 " did you mean 2024.01 and .02? So something happened between 2023.23 and 2024.01 ? That might help us narrow this down.

You are correct, I meant 2024.01 and 2024.02 releases were affected.  Thanks for the catch.

Joe

**Comment by RoelHermus** [comment]
I've the same problem with the 2024 versions. But there is more. When I open a new sequence and I click on the effect bar, with the missing icons, Xlichts crashed. In the version 2024.2 and .3 . I use Ubuntu with one screen. I'm back to the 2023.23 version.

**Comment by RoelHermus** [comment]
Thanks for your responce. This is the zip file:
[xLights_dbgrpt-6730-20240214T162721.zip](https://github.com/xLightsSequencer/xLights/files/14281709/xLights_dbgrpt-6730-20240214T162721.zip)

**Comment by RoelHermus** [comment]
![Schermafdruk van 2024-02-14 16-39-35](https://github.com/xLightsSequencer/xLights/assets/32022225/4fd4ef19-c23e-4470-9924-0ef504f7159c)

I click on the effect bar, the effect setting windows is updated, on the effect bar  the icon showed and an error is rized

**Comment by derwin12** [comment]
I can duplicate this in linux as well. Let me see if I can code around this.
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/c35a1df2-29b8-426f-a9e2-4b5c9a679911)

**Comment by derwin12** [comment]
This is a bigger, core issue. All bitmap buttons are clipped. It seems that wxWidgets and/or gtk is adding borders to the bitmaps. So every icon in xlights is messed under Linux.
Missing lock icons
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/20da482a-b0f7-4260-8417-26776b0865f6)
Delete and Save icons
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/56a63e9f-4ade-44d1-9bee-f5813cb31974)

**Comment by derwin12** [comment]
@dkulp Anything that can be done around this? 
"Patch for buttons on Linux having too large of a border. Affects xLights effect toolbar/panel"
https://github.com/xLightsSequencer/wxWidgets/commit/955fe9b6b40752ffa4a4f89d894f926ecf00c609

**Comment by derwin12** [comment]
I have a few lines that make this work. I cant push PR to wxwidgets.
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/7e3aad97-3aaa-436b-b5fb-54c7ea2c7dde)
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/5fd4ad20-abe9-47ef-9e74-321ddc4ce7f5)
![image](https://github.com/xLightsSequencer/xLights/assets/4643499/55fa75b7-79cd-4d93-85ef-6d629e387a3f)

![image](https://github.com/xLightsSequencer/xLights/assets/4643499/f35a10a4-b203-4265-b1b8-6f2ca531be8e)

**Comment by RoelHermus** [comment]
It looks that the problem is solved with version xLights-2024.05-x86_64.AppImage. Thanks you very much, I am happy.

**Comment by Gilimi-the-Grim** [comment]
version 2024.05 looks to be working for me too.  Thank you VERY much for the fix.

-Joe

**Comment by derwin12** [comment]
I was trying to verify the fix worked in .5.. Thanks for the confirmation!
